Update dependencies

willkg opened the pull request:
This updates the install and dev dependencies to recent versions of
things.

willkg commented on setup.py:
```diff
@@ -13,9 +13,9 @@ def read_file(name):
-    "google-cloud-bigquery==1.27.2",
+    "requests==2.25.0",
+    "ujson==4.0.1",
+    "google-cloud-bigquery==2.5.0",
```
We shouldn't be pinning these to specific versions but should instead do ranges or minimum support or something like that.

This might be better?:

```
requests>=2.24.0<3.0.0
ujson>=4.0.0<5.0.0
google-cloud-bigquery>=2.5.0<3.0.0
```

I'm guessing wildly on the version promises for those projects and assuming that they're semver-ish.

wlach replied:
Yes, definitely. Feel free to change that as part of this PR, if you like.

Ben-Wu replied:
Sounds good to me

willkg commented:
@Ben-Wu, @wlach Can you double-check this? I created a virtualenv, installed the bits, ran `bin/run_tests.sh`, and then ate a cookie. Is there more we want to test here?

willkg commented:
Oh! Also, I want to put the bigquery stuff into an extras block. No one needs that unless they're using it like we're going to use it for crash pings.

Ben-Wu reviewed (approved):
Not super important but I think the url and maintainer in setup.py should be changed.

willkg commented:
I fixed the dependencies in `setup.py` so they're not pinned to specific versions, but rather to ranges.

I moved the `google-cloud-bigquery` dependency to `extras_require` so you have to install it with `[bq]` now. I updated the docs accordingly as well as the `Dockerfile`.

I decided we should have a release process written down, so I wrote that down. In doing that, I copied other libraries I work on (Markus, Everett, siggen) because it seemed easier than inventing a new one.

I updated metadata in `setup.py` so that should be "better" now.

willkg commented:
^ @wlach @Ben-Wu Is this ok with you both?
